.github/workflows/sizeComparison.yml

```diff
@@ -0,0 +1,66 @@
+name: sizeComparison
+
+on:
+  push:
+    branches:
+      - main*
+      - fluent2*
+  pull_request:
+    branches:
+      - main*
+      - fluent2*
+
+jobs:
+  archiveHeadBuild:
+    runs-on: macos-12
+    strategy:
+      fail-fast: false
+    steps: 
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+    - name: Switch to current version of Xcode
+      run: scripts/xcode_select_current_version.sh
+    - name: scripts/xcodebuild_wrapper.sh 'ios_device_build Demo.Development Release build -archivePath $GITHUB_WORKSPACE/headBuild archive'
+      run: scripts/xcodebuild_wrapper.sh ios_device_build Demo.Development Release build -archivePath $GITHUB_WORKSPACE/${{ github.event.pull_request.head.sha }} archive
```

Comment:
does this trigger another build? or can we leverage the matrix build triggered by ci.yml?

Reply:
This currently triggers another build. We might be able to add a dependency on ci.yml, and move the uploading of the artifact there, but I'm not sure.

Reply:
Though, if we don't do the build here, we'd probably lose the guarantee that we'd actually have a build from the base hash. I'm not sure how much control we have over a the lifetime of an artifact.

Reply:
what does the build artifact name looks like? how do we guarentee its lifecycle and its uniqueness per PR trigger (and same pr having multiple commit push/iteration)

Reply:
The artifact name is set in the upload-artifact action, and I'm just using the commit hash as the name. So each trigger will produce two, one for the base hash and one for the head hash. We could probably streamline it by making it check for if the base one already exists, but I hadn't looked in to how to actually do that.

Reply:
can you also add comments on what the exact output artifect would look like per steps?